src/coreclr/jit/morph.cpp

```diff
@@ -17440,11 +17440,14 @@ void Compiler::fgRetypeImplicitByRefArgs()
                 newVarDsc->lvAddrExposed     = varDsc->lvAddrExposed;
                 newVarDsc->lvDoNotEnregister = varDsc->lvDoNotEnregister;
 #ifdef DEBUG
-                newVarDsc->lvLclBlockOpAddr   = varDsc->lvLclBlockOpAddr;
-                newVarDsc->lvLclFieldExpr     = varDsc->lvLclFieldExpr;
-                newVarDsc->lvVMNeedsStackAddr = varDsc->lvVMNeedsStackAddr;
-                newVarDsc->lvLiveInOutOfHndlr = varDsc->lvLiveInOutOfHndlr;
-                newVarDsc->lvLiveAcrossUCall  = varDsc->lvLiveAcrossUCall;
+                newVarDsc->lvLclBlockOpAddr        = varDsc->lvLclBlockOpAddr;
+                newVarDsc->lvLclFieldExpr          = varDsc->lvLclFieldExpr;
+                newVarDsc->lvVMNeedsStackAddr      = varDsc->lvVMNeedsStackAddr;
+                newVarDsc->lvLiveInOutOfHndlr      = varDsc->lvLiveInOutOfHndlr;
+                newVarDsc->lvSingleDef             = varDsc->lvSingleDef;
+                newVarDsc->lvSingleDefRegCandidate = varDsc->lvSingleDefRegCandidate;
+                newVarDsc->lvSpillAtSingleDef      = varDsc->lvSpillAtSingleDef;
+                newVarDsc->lvLiveAcrossUCall       = varDsc->lvLiveAcrossUCall;
```

Comment:
These new fields (as well as the existing `lvLiveInOutOfHndlr`) are not DEBUG only, so seems like we should be copying them over always, not just under DEBUG. It may not matter in practice as these fields likely won't have been set when this gets called, but it seems like we ought to be logically consistent.

And we're not copying the DEBUG only `lvSingleDefDisqualifyReason`.

Seems odd we don't have a general local var cloning-type operation we can just call here.

Reply:
Done.